### DISCUSSION TITLE
Booting windows installer from iSCSI

### DISCUSSION THREAD

**Discussion by Ridddick** [ipxe/ipxe#214] [Q&A]
Hi! Having a strange behaviour during windows install from an iSCSI target. 
1. I have attached the iscsi-target to my PC and imaged a win10.iso right onto it. Detaching target from pc.
2. Boot ipxe from another pc and run a config:
```
set netX/gateway 0.0.0.0
set root-path ${base-iscsi}:win10
sanboot ${root-path}
```
3. Windows installation starts normaly, i choose locale, then press install button... Suddenly windows tells me that it can't find controller drivers and offers to search them on USB etc. This is typical pc config ang runnig install from usb goes normaly. Next interesting things are happening, Pressing Browse button i can see an empty local drive so windows does see it. I also can see my target's drive with windows installation files. I'm closing all windows but not restarting. I'm choosing a recovery option and running cmd. Next i'm running D:\setup.exe (target drive) and... it continues to install - choose partition and so on!

What's going on??

Update.
Actualy windows is asking for a source-drive driver! So it can't work with iSCSI after ipxe has transfered boot to windows? Is there a workaround?

**Comment by NiKiZe** [comment]
That dialog is often a symptom of setup not finding the installation source.

Instead of running the setup that is started by default, try to open CMD shift+f5 go to your real installation source (the iSCSI device) and run setup.exe from there.

Hopefully by doing so you will find the real issue along the way.

**Reply by Ridddick** [reply]
You got ahead of me )) But i'm actualy doing exactly te same, manualy run setup from cmd. So what's the solution? If it's a persistent problem I guess the only way is to use wimboot and run setup.bat. But i didn't find how to run wimboot with boot.wim that is on iSCSI target. Can you help me with that?

**Reply by NiKiZe** [reply]
Please double check that it is the installation source, that is the sources directory where the install.wim is located.

use wimboot, you do not load boot.wim from iscsi , use http preferably

**Comment by sebaxakerhtc** [comment]
>     1. Suddenly windows tells me that it can't find controller drivers and offers to search them on USB etc.

You should to add iscsi support to boot.wim and before starting installation be sure you started the iscsi service (net start msiscsi) otherwise boot.wim can't find the driver for install media.
And also, i recomend to use the --keep option with sanboot command.

**Comment by mcb30** [comment]
Do not attempt to boot the Windows installation ISO from an iSCSI disk.  You can (with difficulty) make it work, but it will be far easier to just use https://ipxe.org/wimboot instead to load `boot.wim` via HTTP.
